tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc

```diff
@@ -397,6 +493,36 @@ REGISTER_KERNEL_BUILDER(
     Name(PREFIX_OP_NAME(CuckooHashTableInsert)).Device(DEVICE_GPU),
     HashTableInsertGpuOp);
 
+// Table accum op.
+template <class K, class V>
+class HashTableAccumGpuOp : public OpKernel {
+ public:
+  explicit HashTableAccumGpuOp(OpKernelConstruction* ctx) : OpKernel(ctx) {}
+
+  void Compute(OpKernelContext* ctx) override {
+    lookup::LookupInterface* table;
+    OP_REQUIRES_OK(ctx, GetLookupTable("table_handle", ctx, &table));
+    core::ScopedUnref unref_me(table);
+    lookup::CuckooHashTableOfTensorsGpu<K, V>* table_cuckoo =
+        (lookup::CuckooHashTableOfTensorsGpu<K, V>*)table;
+
+    DataType expected_input_0 =
+        (ctx->input_dtype(0) == DT_RESOURCE) ? DT_RESOURCE : DT_STRING_REF;
```

Comment:
In OP registration, `input(0)` is always resource. No need to discriminate `DT_RESOURCE` and `DT_STRING_REF`